packages/components/src/AtlantisThemeContext/AtlantisThemeContext.tsx

```diff
@@ -0,0 +1,129 @@
+import { darkTokens, tokens } from "@jobber/design";
+import React, {
+  PropsWithChildren,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import merge from "lodash/merge";
+import {
+  AtlantisThemeContextProviderProps,
+  AtlantisThemeContextValue,
+  THEME_CHANGE_EVENT,
+  Theme,
+} from "./types";
+import { useThemeContextEventQueue } from "./useThemeContextEventQueue";
+import styles from "./AtlantisThemeContext.css";
+
+export const atlantisThemeContextDefaultValues: AtlantisThemeContextValue = {
+  theme: "light",
+  tokens: tokens,
+};
+
+const actualDarkTokens = merge({}, tokens, darkTokens);
+
+const AtlantisThemeContext = createContext(atlantisThemeContextDefaultValues);
+
+export function AtlantisThemeContextProvider({
+  children,
+  dangerouslyForceThemeForProvider,
+}: AtlantisThemeContextProviderProps) {
+  if (dangerouslyForceThemeForProvider) {
+    return (
+      <InternalStaticThemeProvider
+        dangerouslyForceThemeForProvider={dangerouslyForceThemeForProvider}
+      >
+        {children}
+      </InternalStaticThemeProvider>
+    );
+  }
+
+  return (
+    <InternalDynamicThemeProvider>{children}</InternalDynamicThemeProvider>
+  );
+}
+
+function InternalDynamicThemeProvider({ children }: PropsWithChildren) {
+  const initialTheme: Theme =
+    (globalThis.document.documentElement.dataset.theme as Theme) ?? "light";
+
+  const [internalTheme, setInternalTheme] = useState<Theme>(initialTheme);
+  const currentTokens = initialTheme === "dark" ? actualDarkTokens : tokens;
+
+  const {
+    isEmpty,
+    dequeueThemeChange,
+    themeChangeQueue,
+    handleThemeChangeEvent,
+  } = useThemeContextEventQueue();
+
+  useEffect(() => {
+    if (!globalThis.window) return;
+    globalThis.window.addEventListener(
+      THEME_CHANGE_EVENT,
+      handleThemeChangeEvent,
+    );
+
+    return () => {
+      if (!globalThis.window) return;
```

Comment:
I think so to as I think about it more. Since the SSR portion wouldn't add the event listener (and return this callback) I don't think this line is needed either
Removed in: 07ea4ca89b7c60e20b0fe74d9ada9cff17a55550